integration/test/Test/Errors.hs

```diff
@@ -0,0 +1,54 @@
+{-# OPTIONS -Wno-ambiguous-fields #-}
```

Comment:
The problem with that warning is that there is no way to get rid of it if you have both fields in scope. My reasoning for why it's ok to suppress it is that by the time it becomes an error, presumably we'll also be given a way to resolve it.

Reply:
Yeah, either by qualifying or hiding the excess field. It has been decided that there will be no type based disambiguation because the way it is supposed to work in GHC is that types follow from names, never vice versa.